bgpd/bgp_evpn.c

```diff
@@ -2164,26 +2164,24 @@ static int delete_global_type2_routes(struct bgp *bgp, struct bgpevpn *vpn)
 
 	rddest = bgp_node_lookup(bgp->rib[afi][safi],
 				 (struct prefix *)&vpn->prd);
-	if (rddest && bgp_dest_has_bgp_path_info_data(rddest)) {
-		table = bgp_dest_get_bgp_table_info(rddest);
-		for (dest = bgp_table_top(table); dest;
-		     dest = bgp_route_next(dest)) {
-			const struct prefix_evpn *evp =
-				(const struct prefix_evpn *)bgp_dest_get_prefix(
-					dest);
+	if (!rddest)
+		return 0;
 
-			if (evp->prefix.route_type != BGP_EVPN_MAC_IP_ROUTE)
-				continue;
+	table = bgp_dest_get_bgp_table_info(rddest);
```

Comment:
bgp_dest_get_bgp_table_info can return NULL, right?

Reply:
Done, thanks for the review.